lib/default.nix

```diff
@@ -80,4 +80,16 @@ with haskellLib;
     let f' = if lib.isFunction f then f else import f;
         args' = (builtins.intersectAttrs (builtins.functionArgs f') scope) // args;
     in f' args';
+
+  # Collect all (transitive) Haskell library dependencies of a
+  # component.
+  ## flatLibDepends :: Component -> [Package]
```

Comment:
This is pretty cool. Is that identical to the walking in the `component-builder`?

With this we could build a simple `Component -> [License]` collector as well I think. Just map the license key over the packages, and you have the licenses of all dependencies?

Should `components` have a circular link back to their package?

Reply:
Yes, this function was originally inside `comp-builder.nix`. But I needed to pull it out into the library to re-use it.

I have not yet seen a need to link the component back to its package.

Reply:
Should we replace the one in the `comp-builder.nix` with a call to this then?

Reply:
Yes, it's already gone. I have made special efforts not to duplicate builder logic.

Reply:
Ahh, I missed the change in #121. Sorry!